### ISSUE TITLE
CouchDB RAS incorrectly terminating

### ISSUE THREAD

**Issue by hbunchgithub** [galasa-dev/projectmanagement#560]
Requested by Mike to track issue found on VTP Eco.

**Comment by Michael-Baylis** [comment]
I have discovered the following message:-
19/01/2021 13:58:24.926 INFO d.g.r.c.i.CouchdbRasWriteByteChannel - Stored artifact /zosBatchJobs/createPBFile/GALXUMFY_supplied_JCL, length=575, contentType=plain/text
19/01/2021 13:58:24.939 ERROR d.g.r.c.i.CouchdbRasWriteByteChannel - The run document has been updated by another engine, terminating now to avoid corruption

For some reason I believe a timing issue on this test is causing the run instance master document to be updated by multiple threads at the same time,  so one of the threads detects the "outside" update and terminates the run immediately for integrity reasons.

Need to figure out what threads could be updating the master document and syncronise them.

For some reason,  this run then gets into a reset loop.      it terminates,   resman detects it is lost and resets it,     it runs and eventualy hits the same problem and terminates,   resman resets,  etc etc etc
